addons/core/functions/fnc_radioToRequestCount.sqf

```diff
@@ -23,44 +23,36 @@
         (false call TFAR_fnc_radioToRequestCount) params ["_radiosToReplace","_TF_SettingsToCopy"];
 */
 
+params[["_allRadios", false, [true]]];
 private _to_remove = [];
-private _allRadios = _this;
-
-private _classes = call TFAR_fnc_getDefaultRadioClasses;
-private _personalRadio = _classes select 1;
-private _riflemanRadio = _classes select 2;
-private _defaultRadio = _riflemanRadio;
-
-if ((TFAR_givePersonalRadioToRegularSoldier) or {leader TFAR_currentUnit == TFAR_currentUnit} or {rankId TFAR_currentUnit >= 2}) then {
-    _defaultRadio = _personalRadio;
-};
-
 private _TF_settingsToCopy = [];
+
 {
     if (_x call TFAR_fnc_isPrototypeRadio) then {
         _to_remove pushBack _x;
-        TFAR_RadioReqLinkFirstItem = true;
+        _TF_settingsToCopy pushBack "";
     } else {
         if (_x call TFAR_fnc_isRadio) then {
             if ((_x call TFAR_fnc_getRadioOwner) == "") then {
                 [_x, getPlayerUID player] call TFAR_fnc_setRadioOwner;
             };
             if (((_x call TFAR_fnc_getRadioOwner) != (getPlayerUID player)) or _allRadios) then {
                 _to_remove pushBack _x;
-                _TF_settingsToCopy set [0, _x];
-                TFAR_RadioReqLinkFirstItem = true;
+                _TF_settingsToCopy pushBack _x;
             };
         };
     };
-    true;
+    nil
 } count (assignedItems TFAR_currentUnit);
 
-private _uniqueItems = (items TFAR_currentUnit);
-_uniqueItems = _uniqueItems arrayIntersect _uniqueItems;//Remove duplicates //#BUG this causes that only one Radio will be requested per request if you have multiple with same classname
+private _allItems = ((getItemCargo (uniformContainer TFAR_currentUnit)) select 0);
+_allItems append ((getItemCargo (vestContainer TFAR_currentUnit)) select 0);
+_allItems append ((getItemCargo (backpackContainer TFAR_currentUnit)) select 0);
```

Comment:
You still remove have duplicates ;)

Reply:
No, i don't want to remove duplicates.
I want to replace every radio. So if the unit has 2 Prototypes of the same kind, the unit will receive 2 unique radios instead.
Anyways I'll optimize the evaluation speed in that case, where the unit isn't carrying a radio needing to be replaced.